### PR TITLE
Made a new admin verb, Send Fax, which lets admins send a fax without having to reply to one first.

### PR DESCRIPTION
**Pull request by sonime**
Was requested by MrPain666 and a good coding lesson.

### CODE REVIEW

File: html/changelogs/DeadPoolJ.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+author: DeadPoolJ
+changes: 
+	- rscadd: Made a new admin verb, Send Fax, which lets admins send a fax without having to reply to one first.
```
Comment (N3X15) [inline html/changelogs/DeadPoolJ.yml:3]: THE FUCK DID I SAY ABOUT TABS

**Comment by sonime** [comment]
FUCK I'll fix it tomorrow

File: html/changelogs/DeadPoolJ.yml (hunk 1 above)
Comment (clusterfack) [inline html/changelogs/DeadPoolJ.yml]: Holy fuck it turned red, what magic is this

Comment (PJB3005) [inline html/changelogs/DeadPoolJ.yml:3]: Reason number 1 why YAML is shit: it enforces soft tabs.

File: code/modules/admin/admin_verbs.dm [hunk 2]
```diff
@@ -1042,3 +1043,41 @@ var/list/admin_verbs_mod = list(
...
+	set name = "Send Fax"
+	set desc = "Sends a fax to all fax machines."
+
+	var/cntcom_msg = input(src, "Please enter a message to send via secure connection. NOTE: BBCode does not work, but HTML tags do! Use <br> for line breaks.", "Outgoing message from Centcomm", "") as message|null
+	if(!cntcom_msg)	return
+
+	var/customname = input(src, "Pick a title for the report", "Title") as text|null
+
+	for(var/obj/machinery/faxmachine/F in allfaxes)
+		if(! (F.stat & (BROKEN|NOPOWER) ) )
```
Comment (PJB3005) [inline code/modules/admin/admin_verbs.dm:1058]: `if(F.stat ^ (BROKEN | NOPOWER))` will have the same result, but it's cleaner.

File: code/modules/admin/admin_verbs.dm [hunk 3]
```diff
@@ -1042,3 +1043,41 @@ var/list/admin_verbs_mod = list(
...
+
+			// animate! it's alive!
+			flick("faxreceive", F)
```
Comment (PJB3005) [inline code/modules/admin/admin_verbs.dm:1061]: You should probably make this into a proc because right now this is copy-pasted from the fax machine code.

**Comment by sonime** [comment]
Make the while thing a proc? It's not already? What do you mean?

**Comment by PJB3005** [comment]
I mean, make the whole code for actually printing a fax a proc on the faxing machine, you can then call the proc like any other. Making it a proc removes copy paste and makes it more maintainable (no need to chase 3 different files to edit the faxing machin's function).

Procs are defined like this:

``` DM
/type/of/object/proc/procname(var/argument1, var/argument2, var/argument3...)
    // Code goes here...
```

**Comment by mrpain666** [comment]
RIP.

Dont stop trying I believe in you!

**Comment by sonime** [comment]
Looking in fax.dm there is a proc called SendFax.

**Comment by unid15** [comment]
PJB means that the highlighted part should be turned into a proc (something like obj/machinery/faxmachine/proc/ReceiveFax(sent, sentname, mob/sender, dpt))

Then you'd just do

```
for(var/obj/machinery/faxmachine/F in allfaxes)
    F.ReceiveFax(..................)
```

when you want to send a fax to all fax machines

**Comment by sonime** [comment]
The problem is that it sends it to a specific department, not all faxes.

**Comment by mrpain666** [comment]
awww

**Comment by sonime** [comment]
Spoilers: I started over somewhat. Its on the way fam.

**Pull request closed without merging**